File: songs/ocaml/mysongs.ml
type tl_tags  = string list;;
type tl_song  = {
  title : string; tags : tl_tags option;
};;
type tl_songs = tl_song list;;

let songs: tl_songs = [
    { title = "Cantaloupe Island";
      tags = Some ["60s"; "jazz"]
    };
    { title = "Let it Be";
      tags = Some ["60s"; "rock"]
    };
    { title = "Knockin' on Heaven's Door";
      tags = Some ["70s"; "rock"]
    };
    { title = "Emotion";
      tags = Some ["70s"; "pop"]
    };
    { title = "The River";
      tags = None
    }
];;

File: songs/rust/my_songs.rs
use std::fmt;

pub struct Song<'lt> {
    title: &'lt str,
    pub tags : Option<Vec<&'lt str>>
}

impl<'lt> fmt::Display for Song<'lt> {
    fn fmt(&self, f: &mut fmt::Formatter)-> fmt::Result {
        write!(f, "({}, {:?})", self.title, self.tags)
    }
}

pub fn get_songs<'lt>() -> Vec<Song<'lt>> {
    vec![
        Song {  title: "Cantaloupe Island",
                tags: Some(vec!["60s", "jazz"])},
        Song {  title: "Let It Be",
                tags: Some(vec!["60s", "rock"])},
        Song {  title: "Knockin' on Heaven's Door",
                tags: Some(vec!["70s", "rock"])},
        Song {  title: "Emotion",
                tags: Some(vec!["70s", "pop"])},
        Song {  title: "The River", tags: None }
    ]
}
